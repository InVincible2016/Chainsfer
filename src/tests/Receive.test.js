import until from 'enzyme-shallow-until'
import ShallowWrapper from 'enzyme/ShallowWrapper'

import Receive from '../components/ReceiveComponent'
import ReceiveWalletSelection from '../containers/ReceiveWalletSelectionContainer'

ShallowWrapper.prototype.until = until

describe('ReceiveComponent', () => {
  const history = { location: '/receive' }

  it('step = 1', () => {
    const wrapper = shallow(
      <Receive step={1} history={history} />
    ).until(ReceiveWalletSelection)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('step = 2', () => {
    const wrapper = shallow(
      <Receive step={2} history={history} />
    ).until(ReceiveWalletSelection)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('step = 3', () => {
    const wrapper = shallow(
      <Receive step={3} history={history} />
    ).until(ReceiveWalletSelection)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('step = 4', () => {
    const wrapper = shallow(
      <Receive step={4} history={history} />
    ).until(ReceiveWalletSelection)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
