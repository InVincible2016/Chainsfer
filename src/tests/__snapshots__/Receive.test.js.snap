// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReceiveComponent step = 1 1`] = `
<div
  className="MuiGrid-container-3 MuiGrid-direction-xs-column-6"
>
  <WithStyles(Grid)
    item={true}
  >
    <WithStyles(MyStepper)
      actionType="receive"
      step={0}
    />
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
  >
    <WithStyles(Grid)
      alignItems="center"
      className={false}
      container={true}
      direction="column"
    >
      <WithStyles(Grid)
        className="ReceiveComponent-subComponent-2"
        item={true}
      >
        <Connect(ReceivePasswordContainer) />
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</div>
`;

exports[`ReceiveComponent step = 2 1`] = `
<div
  className="MuiGrid-container-3 MuiGrid-direction-xs-column-6"
>
  <WithStyles(Grid)
    item={true}
  >
    <WithStyles(MyStepper)
      actionType="receive"
      step={1}
    />
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
  >
    <WithStyles(Grid)
      alignItems="center"
      className={false}
      container={true}
      direction="column"
    >
      <WithStyles(Grid)
        className="ReceiveComponent-subComponent-2"
        item={true}
      >
        <Connect(ReceiveWalletSelectionContainer) />
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</div>
`;

exports[`ReceiveComponent step = 3 1`] = `
<div
  className="MuiGrid-container-3 MuiGrid-direction-xs-column-6"
>
  <WithStyles(Grid)
    item={true}
  >
    <WithStyles(MyStepper)
      actionType="receive"
      step={2}
    />
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
  >
    <WithStyles(Grid)
      alignItems="center"
      className={false}
      container={true}
      direction="column"
    >
      <WithStyles(Grid)
        className="ReceiveComponent-subComponent-2"
        item={true}
      >
        <Connect(ReceiveReviewContainer) />
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</div>
`;

exports[`ReceiveComponent step = 4 1`] = `
<div
  className="MuiGrid-container-3 MuiGrid-direction-xs-column-6 ReceiveComponent-rootReceipt-1"
>
  <WithStyles(Grid)
    item={true}
  />
  <WithStyles(Grid)
    item={true}
  >
    <WithStyles(Grid)
      alignItems="center"
      container={true}
      direction="column"
    >
      <WithStyles(Grid)
        className="ReceiveComponent-subComponent-2"
        item={true}
      >
        <Connect(ReceiveReceiptContainer) />
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</div>
`;
